recaf-launcher/src/main/java/dev/xdark/recaf/Launcher.java

```diff
@@ -0,0 +1,251 @@
+package dev.xdark.recaf;
+
+import com.google.gson.GsonBuilder;
+import dev.dirs.BaseDirectories;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import joptsimple.OptionParser;
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import org.apache.commons.io.IOUtils;
+
+/**
+ * Recaf's launcher entrypoint.
+ *
+ * @author xDark
+ */
+public final class Launcher {
+
+  private static final String API_URL = "https://api.github.com/repos/Col-E/Recaf/releases/latest";
+  private static final String ISSUES_URL = "https://github.com/Col-E/Recaf/issues";
+
+  private static final List<String> UPDATE_FAIL = Collections.singletonList("--updateFailed");
+  private static final PrintStream PS = System.err;
+
+  /**
+   * Deny public constructions.
+   */
+  private Launcher() {
+  }
+
+  /**
+   * Main entrypoint.
+   *
+   * @param args Program arguments.
+   */
+  public static void main(String[] args) {
+    ReleaseInfo info = null;
+    try {
+      info = fetchReleaseInfo();
+      log("Successfully fetched release info");
+    } catch (IOException ex) {
+      log("Could not fetch release info: ", ex);
+    }
+
+    OptionParser parser = new OptionParser();
+    OptionSpec<Path> jarOption = parser.accepts("home", "Recaf jar file")
+        .withRequiredArg()
+        .withValuesConvertedBy(new PathValueConverter())
+        .defaultsTo(Paths.get(BaseDirectories.get().dataDir).resolve("Recaf").resolve("Recaf.jar"));
```

Comment:
Were we able to replicate if the powershell problem was fixed? I checked upstream and it looks like they don't actually have a fallback value, it will just return `null` instead. We may still want to fallback on the environment variable.